### ISSUE TITLE
Update selection

### ISSUE THREAD

**Issue by smebberson** [vscodeshift/function-codemorphs#10]
Great plugin!

It would be fantastic if the plugin would also update the selection after the codemorph such that the entire function remained selected.

**Comment by jedwards1211** [comment]
Hadn't thought about that, makes sense.  I'll have to think for a bit about how to do it in a robust way...right now it's replacing the entire document text even though it's only modifying the function AST node, and I'm not 100% sure how to determine the location of the transformed function properly.  I guess I can probably reparse the generated code and find the node at the same path to get its new printed location

**Comment by smebberson** [comment]
Ah, that makes sense why there's a little delay (because it's replacing the entire document).

**Comment by jedwards1211** [comment]
If I required the selection to contain just the function, then it would be easy enough to parse just the selection. But it's designed to work with the cursor anywhere inside a function too, which is how I usually use it.

The most significant delay (the first time you use it within a give project since launch) should be loading the babel parser and your babel config, if applicable

**Comment by smebberson** [comment]
I see, using it with the cursor anywhere within the function would be ideal. I did actually try that but it didn't work, although, I think I had my cursor on the `const` statement, and then I tried the variable name, but not the function itself. I'll try that.

**Comment by jedwards1211** [comment]
Yeah, the cursor has to be anywhere between the start and end of the AST node for the function.  For arrow functions that starts with the `(` and ends at the end of the body expression or `}` if the body is a block statement, and doesn't include any preceeding variable declarator.  Now that I think about it though, I could support cursor in the preceeding variable declarator too!

**Comment by jedwards1211** [comment]
okay, I just published release v1.2.0, which should work with the cursor on the `const` (as long as you're not declaring multiple variables separated by commas)
